### PR TITLE
Add Node.js runtime metrics dashboards

### PR DESCRIPTION
**Pull request by Qard**
Depends on elastic/apm-agent-nodejs#1021
Closes elastic/apm-agent-nodejs#993

### CODE REVIEW

File: apm-agent-nodejs/dashboards/nodejs-runtime-metrics-6.x.json [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      }
+    }
+  },
+  {
+    "_id": "582c9670-825c-11e9-b6c5-07b7c21e0d7c",
+    "_type": "visualization",
+    "_source": {
+      "title": "[nodejs] Event Loop Delay",
+      "visState": "{\"title\":\"[nodejs] Event Loop Delay\",\"type\":\"metrics\",\"params\":{\"id\":\"61ca57f0-469d-11e7-af02-69e470af7417\",\"type\":\"timeseries\",\"series\":[{\"id\":\"61ca57f1-469d-11e7-af02-69e470af7417\",\"color\":\"#68BC00\",\"split_mode\":\"everything\",\"metrics\":[{\"id\":\"61ca57f2-469d-11e7-af02-69e470af7417\",\"type\":\"avg\",\"field\":\"nodejs.eventloop.delay.ms\"}],\"separate_axis\":0,\"axis_position\":\"right\",\"formatter\":\"ms,ms,2\",\"chart_type\":\"line\",\"line_width\":1,\"point_size\":1,\"fill\":0.5,\"stacked\":\"none\",\"label\":\"Event Loop Delay\"}],\"time_field\":\"@timestamp\",\"index_pattern\":\"\",\"interval\":\"1m\",\"axis_position\":\"left\",\"axis_formatter\":\"number\",\"axis_scale\":\"normal\",\"show_legend\":1,\"show_grid\":1,\"default_index_pattern\":\"apm-*\",\"filter\":\"processor.event: \\\"metric\\\", context.service.agent.name: \\\"nodejs\\\"\"},\"aggs\":[]}",
```
Comment (watson) [inline apm-agent-nodejs/dashboards/nodejs-runtime-metrics-6.x.json]:
```suggestion
      "visState": "{\"title\":\"[nodejs] Event Loop Delay\",\"type\":\"metrics\",\"params\":{\"id\":\"61ca57f0-469d-11e7-af02-69e470af7417\",\"type\":\"timeseries\",\"series\":[{\"id\":\"61ca57f1-469d-11e7-af02-69e470af7417\",\"color\":\"#68BC00\",\"split_mode\":\"everything\",\"metrics\":[{\"id\":\"61ca57f2-469d-11e7-af02-69e470af7417\",\"type\":\"avg\",\"field\":\"nodejs.eventloop.delay.avg.ms\"}],\"separate_axis\":0,\"axis_position\":\"right\",\"formatter\":\"ms,ms,2\",\"chart_type\":\"line\",\"line_width\":1,\"point_size\":1,\"fill\":0.5,\"stacked\":\"none\",\"label\":\"Event Loop Delay\"}],\"time_field\":\"@timestamp\",\"index_pattern\":\"\",\"interval\":\"1m\",\"axis_position\":\"left\",\"axis_formatter\":\"number\",\"axis_scale\":\"normal\",\"show_legend\":1,\"show_grid\":1,\"default_index_pattern\":\"apm-*\",\"filter\":\"processor.event: \\\"metric\\\", context.service.agent.name: \\\"nodejs\\\"\"},\"aggs\":[]}",
```

File: apm-agent-nodejs/dashboards/nodejs-runtime-metrics-7.x.json [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      "visualization": "7.0.1"
+    },
+    "_references": []
+  },
+  {
+    "_id": "582c9670-825c-11e9-b6c5-07b7c21e0d7c",
+    "_type": "visualization",
+    "_source": {
+      "title": "[nodejs] Event Loop Delay",
+      "visState": "{\"title\":\"[nodejs] Event Loop Delay\",\"type\":\"metrics\",\"params\":{\"id\":\"61ca57f0-469d-11e7-af02-69e470af7417\",\"type\":\"timeseries\",\"series\":[{\"id\":\"61ca57f1-469d-11e7-af02-69e470af7417\",\"color\":\"#68BC00\",\"split_mode\":\"everything\",\"metrics\":[{\"id\":\"61ca57f2-469d-11e7-af02-69e470af7417\",\"type\":\"avg\",\"field\":\"nodejs.eventloop.delay.ms\"}],\"separate_axis\":0,\"axis_position\":\"right\",\"formatter\":\"ms,ms,2\",\"chart_type\":\"line\",\"line_width\":1,\"point_size\":1,\"fill\":0.5,\"stacked\":\"none\",\"label\":\"Event Loop Delay\"}],\"time_field\":\"@timestamp\",\"index_pattern\":\"\",\"interval\":\"1m\",\"axis_position\":\"left\",\"axis_formatter\":\"number\",\"axis_scale\":\"normal\",\"show_legend\":1,\"show_grid\":1,\"default_index_pattern\":\"apm-*\",\"filter\":\"processor.event: \\\"metric\\\", context.service.agent.name: \\\"nodejs\\\"\"},\"aggs\":[]}",
```
Comment (watson) [inline apm-agent-nodejs/dashboards/nodejs-runtime-metrics-7.x.json]:
```suggestion
      "visState": "{\"title\":\"[nodejs] Event Loop Delay\",\"type\":\"metrics\",\"params\":{\"id\":\"61ca57f0-469d-11e7-af02-69e470af7417\",\"type\":\"timeseries\",\"series\":[{\"id\":\"61ca57f1-469d-11e7-af02-69e470af7417\",\"color\":\"#68BC00\",\"split_mode\":\"everything\",\"metrics\":[{\"id\":\"61ca57f2-469d-11e7-af02-69e470af7417\",\"type\":\"avg\",\"field\":\"nodejs.eventloop.delay.avg.ms\"}],\"separate_axis\":0,\"axis_position\":\"right\",\"formatter\":\"ms,ms,2\",\"chart_type\":\"line\",\"line_width\":1,\"point_size\":1,\"fill\":0.5,\"stacked\":\"none\",\"label\":\"Event Loop Delay\"}],\"time_field\":\"@timestamp\",\"index_pattern\":\"\",\"interval\":\"1m\",\"axis_position\":\"left\",\"axis_formatter\":\"number\",\"axis_scale\":\"normal\",\"show_legend\":1,\"show_grid\":1,\"default_index_pattern\":\"apm-*\",\"filter\":\"processor.event: \\\"metric\\\", context.service.agent.name: \\\"nodejs\\\"\"},\"aggs\":[]}",
```

File: apm-agent-nodejs/dashboards/nodejs-runtime-metrics-6.x.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      "kibanaSavedObjectMeta": {
+        "searchSourceJSON": "{\"query\":{\"query\":\"\",\"language\":\"lucene\"},\"filter\":[]}"
+      }
+    }
+  },
+  {
+    "_id": "cf361160-825c-11e9-b6c5-07b7c21e0d7c",
+    "_type": "visualization",
+    "_source": {
+      "title": "[nodejs] CPU Category Usage",
```
Comment (watson) [inline apm-agent-nodejs/dashboards/nodejs-runtime-metrics-6.x.json]: What does "Category" mean in this context? Never seen that associated with CPU metrics before

Comment (Qard) [reply apm-agent-nodejs/dashboards/nodejs-runtime-metrics-6.x.json]: System or User CPU Usage. Not sure if there's a better name for it? 🤷‍♂

Comment (axw) [reply apm-agent-nodejs/dashboards/nodejs-runtime-metrics-6.x.json]: Just "CPU Usage" seems fine to me, and more in line with the other visualisations.

**Comment by watson** [comment]
I'm working on an update to this dashboard. So far I've made the following changes (not you commited):

- Changed the unit in "CPU Usage" to be percent
- Added an `ms` suffix to the "Event Loop Delay" graph

So far that looks like this:

image

I'm also considering using different colors for active handles and active requests as it's not possible to see which is which in the graph above when they switch places in the graph. Here's an example of how that would look:

image

**Comment by watson** [comment]
Ended up swapping the colors on the active requests/handles graph, so that active handles get the red color.

**Comment by watson** [comment]
I need to do the same for the 6.x dashboard

**Comment by watson** [comment]
Added the 6.x dashboard as well. Also, forgot to mention previously that I also removed the decimals shown for event loop delay as a sub-millisecond resolution wasn't that important. Better have more screen real-estate.

**Pull request merged**